src/hooks/useCertGen.ts

```diff
@@ -41,7 +42,7 @@ export function useGenerateCertificate({
     }
 
     generateCertificateAndImage();
-  }, [certificateDetails]);
+  }, []);
```

Comment:
but now this will not re-run if certificateDetails are available right?

Reply:
But once we get the certificate details on page reload using useEffect then , i think it should not run useGenerateCertificate , we can use a flag instead of this but then we will need to add a button that will regenerate if the details are not availble , i can add that if you want me to and then the dependency array will have certificateDetails.